packages/sdk/src/server/__tests__/server.test.ts

```diff
@@ -687,4 +656,155 @@ describe("ServerClient", () => {
+      client = new ServerClient(
+        {
+          publicKey: "test-public-key",
+          secretKey: "test",
+          oauth: {
+            clientId: "test-client-id",
+            clientSecret: "test-client-secret",
+          },
+        },
+        oauthClientMock,
+      );
+    });
+
+    it("should include externalUserId and environment headers", async () => {
+      fetchMock.mockResponseOnce(
+        JSON.stringify({
+          result: "workflow-response",
+        }),
+        {
+          headers: {
+            "Content-Type": "application/json",
+          },
+        },
+      );
+
+      const result = await client.invokeWorkflowForExternalUser("https://example.com/workflow", "external-user-id", {
+        body: {
+          foo: "bar",
+        },
+      });
+
+      expect(result).toEqual({
+        result: "workflow-response",
+      });
+
+      expect(fetchMock).toHaveBeenCalledWith(
+        "https://example.com/workflow",
+        expect.objectContaining({
+          headers: expect.objectContaining({
+            "X-PD-External-User-ID": "external-user-id",
+            "X-PD-Environment": "production",
+          }),
+        }),
+      );
+    });
+
+    it("should throw error when externalUserId is missing", async () => {
+      await expect(client.invokeWorkflowForExternalUser("https://example.com/workflow", "", {
+        body: {
+          foo: "bar",
+        },
+      })).rejects.toThrow("External user ID is required");
+    });
+  });
+
+  describe("ServerClient - buildWorkflowUrl", () => {
+    describe("Default domain (m.pipedream.net)", () => {
+      it("should return full URL if input is a full URL with protocol", () => {
+        const input = "https://en123.m.pipedream.net";
+        const expected = "https://en123.m.pipedream.net/";
+        expect(client["buildWorkflowUrl"](input)).toBe(expected);
+      });
+
+      it("should return full URL if input is a URL without protocol", () => {
+        const input = "en123.m.pipedream.net";
+        const expected = "https://en123.m.pipedream.net/";
+        expect(client["buildWorkflowUrl"](input)).toBe(expected);
+      });
+
+      it("should construct URL with 'm.pipedream.net' if input is an endpoint ID", () => {
+        const input = "en123";
+        const expected = "https://en123.m.pipedream.net";
+        expect(client["buildWorkflowUrl"](input)).toBe(expected);
+      });
+
+      it("should handle input with a path in full URL with protocol", () => {
+        const input = "https://en123.m.pipedream.net/foo";
+        const expected = "https://en123.m.pipedream.net/foo";
+        expect(client["buildWorkflowUrl"](input)).toBe(expected);
+      });
+
+      it("should handle input with a path when no protocol is provided", () => {
+        const input = "en123.m.pipedream.net/foo";
+        const expected = "https://en123.m.pipedream.net/foo";
+        expect(client["buildWorkflowUrl"](input)).toBe(expected);
+      });
+    });
```

Comment:
Should probably test for when the wrong protocol is provided other than `https`.

Reply:
we'll accept `http://` actually and then the ALB will upgrade the caller to `https://`, kind of hard to test at this level IMO